lib/rdf/repository.rb

```diff
@@ -195,22 +197,18 @@ def transaction(graph_name = nil, &block)
     # @param  [RDF::Resource] graph_name
     # @return [RDF::Transaction]
     # @since  0.3.0
-    def begin_transaction(graph_name)
-      RDF::Transaction.new(graph: graph_name)
+    def begin_transaction(mutable: false)
+      @tx_class.new(self, mutable: mutable)
     end
 
     ##
     # Rolls back the given transaction.
     #
-    # Subclasses implementing transaction-capable storage adapters may wish
-    # to override this method in order to roll back the given transaction in
-    # the underlying storage.
-    #
     # @param  [RDF::Transaction] tx
     # @return [void] ignored
     # @since  0.3.0
     def rollback_transaction(tx)
```

Comment:
Rollback is now a real thing, and up to the `Transaction` to implement. The base implementation is to simply clear `Transaction#changes`.

Clients are discouraged from trying to re-use rolled back transactions. e.g. for:

```ruby
tx = Transaction.new(repo)
tx.insert(st1)
tx.rollback
tx.insert(st2)
tx.execute
```

Some Transactions may handle this fine, but subclasses for individual repositories are not required to handle it. It's allowable to open a transaction with the upstream repository, kill it on rollback, and provide no mechanism to reopen (or open a new transaction) within the same `Transaction` instance.

If there are ideas about how to lock this down a bit better, I would be happy to hear them.